sandbox_dns_unix.go

```diff
@@ -313,8 +315,31 @@ func (sb *sandbox) rebuildDNS() error {
 	// external v6 DNS servers has to be listed in resolv.conf
 	dnsList = append(dnsList, resolvconf.GetNameservers(currRC.Content, types.IPv6)...)
 
-	// Resolver returns the options in the format resolv.conf expects
-	dnsOptionsList = append(dnsOptionsList, sb.resolver.ResolverOptions()...)
+	// If the user entered a ndots value > 0 remember that. container's resolv.conf
+	// will only have ndots:0 for embedded DNS server to get unqualified names.
+	resOptions := sb.resolver.ResolverOptions()
+	for i, option := range dnsOptionsList {
+		if strings.Contains(option, "ndots") {
+			parts := strings.Split(option, ":")
+			if len(parts) == 0 || len(parts) > 2 {
+				return fmt.Errorf("invalid ndots option %v", option)
+			}
+			if num, err := strconv.Atoi(parts[1]); err != nil {
+				return fmt.Errorf("invalid ndots option %v", option)
+			} else if num > 0 {
+				sb.ndotsSet = true
+			}
+			for _, resOpt := range resOptions {
+				if strings.Contains(resOpt, "ndots") {
```

Comment:
I need some help with understanding this logic. resOptions seems to be statically defined as `ndots:0` (via ResolverOptions()). So, it seems like a very predictable for loop with a deterministic `goto` that follows. It is very hard to understand why we have this on the first place.

Even if we do this a generic mechanism to handle ResolverOptions(), then we need a way to cleanly break out of outer for loop as well. infact I would suggest moving the `goto dnsOptions` outside of this inner for-loop and that will make it bit more readable.